app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java

```diff
@@ -37,6 +38,24 @@
     private EditText mPortEdit;
     private TextView mResultText;
 
+    public final int getPort() {
+        String port = mPortEdit.getText().toString();
+        if (TextUtils.isEmpty(port)) {
+            Toast.makeText(this, "You did not enter a port", Toast.LENGTH_SHORT).show();
```

Comment:
How about throw an exception here, and have the caller show the toast in `catch { ... }`.
Then `notEmptyHostAndPort` becomes unnecessary.